src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/JunctionTreeLinkedDeBruinGraph.java

```diff
@@ -0,0 +1,724 @@
+import org.broadinstitute.hellbender.tools.walkers.haplotypecaller.Kmer;
+import org.broadinstitute.hellbender.tools.walkers.haplotypecaller.graphs.*;
+import org.broadinstitute.hellbender.utils.BaseUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.PrintStream;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Experimental version of the ReadThreadingGraph with support for threading reads to generate JunctionTrees for resolving
+ * connectivity information at longer ranges.
+ *
+ * Note that many of the non-DeBruin graph alterations that are made to ReadThreadingGraph are not made here:
+ * - Non-Unique kmers are not duplicated by this graph
+ * - Kmers are not Zipped together to form a SeqGraph
+ * - The reference path is stored in its entirety rather than being calculated on the fly
+ *
+ * For ease of debugging, this graph supports the method {@link #printSimplifiedGraph(File, int)}} which generates a SequenceGraph and
+ * adds the junction trees to the output .dot file.
+ */
+public class JunctionTreeLinkedDeBruinGraph extends ReadThreadingGraphInterface {
+    private static final long serialVersionUID = 1l;
+    private static final MultiDeBruijnVertex SYMBOLIC_END_VETEX = new MultiDeBruijnVertex(new byte[]{'_'});
+    private MultiSampleEdge SYMBOLIC_END_EDGE;
+
+    private Map<MultiDeBruijnVertex, ThreadingTree> readThreadingJunctionTrees = new HashMap<>();
+
+    // TODO should this be constructed here or elsewhere
+    private Set<Kmer> kmers = new HashSet<>();
+
+    public JunctionTreeLinkedDeBruinGraph(int kmerSize) {
+        this(kmerSize, false, (byte)6, 1);
+    }
+
+    /**
+     * Create a new ReadThreadingAssembler using kmerSize for matching
+     * @param kmerSize must be >= 1
+     */
+    JunctionTreeLinkedDeBruinGraph(final int kmerSize, final boolean debugGraphTransformations, final byte minBaseQualityToUseInAssembly, final int numPruningSamples) {
+        super(kmerSize, debugGraphTransformations, minBaseQualityToUseInAssembly, numPruningSamples);
+    }
+
+    /**
+     * Find vertex and its position in seqForKmers where we should start assembling seqForKmers
+     *
+     * @param seqForKmers the sequence we want to thread into the graph
+     * @return the position of the starting vertex in seqForKmer, or -1 if it cannot find one
+     */
+
+    protected int findStartForJunctionThreading(final SequenceForKmers seqForKmers) {
+        for ( int i = seqForKmers.start; i < seqForKmers.stop - kmerSize; i++ ) {
+            final Kmer kmer1 = new Kmer(seqForKmers.sequence, i, kmerSize);
+            if ( kmerToVertexMap.containsKey(kmer1) ) {
+                return i;
+            }
+        }
+
+        return -1;
+    }
+
+    @Override
+    protected void setToInitialState() {
+        // We don't clear pending here in order to later build the read threading
```

Comment:
isn't pending clear at construction time?  This method is only called during construction.